From c2d5741e9aa78fcc935c140fd44ee61a1c0bf24d Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Wed, 11 Dec 2024 12:18:08 -0700
Subject: [PATCH] Use conda-forge siscone for siscone plugin

* Instead of using the Git submodule and building siscone again, use
  the conda-forge siscone.
   - c.f. https://github.com/conda-forge/siscone-feedstock
---
 example/Makefile.am             | 12 ++++++------
 plugins/SISCone/Makefile.am     |  6 +++---
 plugins/SISCone/makefile.static | 20 +++++---------------
 3 files changed, 14 insertions(+), 24 deletions(-)

diff --git a/example/Makefile.am b/example/Makefile.am
index 22a700af..98fcea74 100644
--- a/example/Makefile.am
+++ b/example/Makefile.am
@@ -42,8 +42,8 @@ noinst_PROGRAMS += 03-plugin
 #03_plugin_SHORTNAME = 03
 03_plugin_CXXFLAGS = $(AM_CXXFLAGS) $(CGAL_CPPFLAGS) -I$(srcdir)/../include
 03_plugin_LDADD = ../src/libfastjet.la $(CGAL_LIBS)
-03_plugin_CXXFLAGS += -I$(srcdir)/../plugins/SISCone -I$(srcdir)/../plugins/SISCone/siscone
-03_plugin_LDADD += ../plugins/SISCone/libSISConePlugin.la ../plugins/SISCone/siscone/siscone/libsiscone.la ../plugins/SISCone/siscone/siscone/spherical/libsiscone_spherical.la
+03_plugin_CXXFLAGS += -I$(srcdir)/../plugins/SISCone -I$(PREFIX)
+03_plugin_LDADD += ../plugins/SISCone/libSISConePlugin.la -L$(PREFIX)/lib -lfastjet
 else
 EXTRA_DIST += 03-plugin.cc
 endif
@@ -124,8 +124,8 @@ fastjet_areas_SOURCES = fastjet_areas.cc
 fastjet_areas_CXXFLAGS = $(AM_CXXFLAGS) $(CGAL_CPPFLAGS) -I$(srcdir)/../include
 fastjet_areas_LDADD = ../src/libfastjet.la $(CGAL_LIBS)
 if BUILD_PLUGIN_SISCONE
-fastjet_areas_CXXFLAGS += -I$(srcdir)/../plugins/SISCone -I$(srcdir)/../plugins/SISCone/siscone
-fastjet_areas_LDADD += ../plugins/SISCone/libSISConePlugin.la ../plugins/SISCone/siscone/siscone/libsiscone.la ../plugins/SISCone/siscone/siscone/spherical/libsiscone_spherical.la
+fastjet_areas_CXXFLAGS += -I$(srcdir)/../plugins/SISCone -I$(PREFIX)
+fastjet_areas_LDADD += ../plugins/SISCone/libSISConePlugin.la -L$(PREFIX)/lib -lsiscone
 endif
 
 fastjet_timing_plugins_SOURCES = fastjet_timing_plugins.cc CmdLine.cc
@@ -134,8 +134,8 @@ fastjet_timing_plugins_LDADD = ../src/libfastjet.la $(CGAL_LIBS)
 fastjet_timing_plugins_CXXFLAGS += -I$(srcdir)/../tools
 fastjet_timing_plugins_LDADD += ../tools/libfastjettools.la 
 if BUILD_PLUGIN_SISCONE
-fastjet_timing_plugins_CXXFLAGS += -I$(srcdir)/../plugins/SISCone -I$(srcdir)/../plugins/SISCone/siscone
-fastjet_timing_plugins_LDADD += ../plugins/SISCone/libSISConePlugin.la ../plugins/SISCone/siscone/siscone/libsiscone.la ../plugins/SISCone/siscone/siscone/spherical/libsiscone_spherical.la
+fastjet_timing_plugins_CXXFLAGS += -I$(srcdir)/../plugins/SISCone -I$(PREFIX)
+fastjet_timing_plugins_LDADD += ../plugins/SISCone/libSISConePlugin.la -L$(PREFIX)/lib -lsiscone
 endif
 if BUILD_PLUGIN_CDFCONES
 fastjet_timing_plugins_CXXFLAGS += -I$(srcdir)/../plugins/CDFCones
diff --git a/plugins/SISCone/Makefile.am b/plugins/SISCone/Makefile.am
index a8618452..74b294fb 100644
--- a/plugins/SISCone/Makefile.am
+++ b/plugins/SISCone/Makefile.am
@@ -1,12 +1,12 @@
-SUBDIRS = siscone fastjet .
+SUBDIRS = fastjet .
 
 if MONOLITHIC_PLUGINS
 noinst_LTLIBRARIES = libSISConePlugin.la
 else
 lib_LTLIBRARIES = libSISConePlugin.la
 endif
-libSISConePlugin_la_CXXFLAGS = $(AM_CXXFLAGS) -I$(srcdir) -I$(srcdir)/siscone -I$(srcdir)/../../include -I./siscone
+libSISConePlugin_la_CXXFLAGS = $(AM_CXXFLAGS) -I$(srcdir) -I$(PREFIX)/include -I$(srcdir)/../../include
 libSISConePlugin_la_SOURCES = SISConeBasePlugin.cc SISConePlugin.cc SISConeSphericalPlugin.cc
-libSISConePlugin_la_LIBADD = ../../src/libfastjet.la siscone/siscone/libsiscone.la siscone/siscone/spherical/libsiscone_spherical.la
+libSISConePlugin_la_LIBADD = ../../src/libfastjet.la -L$(PREFIX)/lib -lsiscone -lsiscone_spherical
 
 EXTRA_DIST = makefile.static
diff --git a/plugins/SISCone/makefile.static b/plugins/SISCone/makefile.static
index ffc01390..d2a74ac2 100644
--- a/plugins/SISCone/makefile.static
+++ b/plugins/SISCone/makefile.static
@@ -6,32 +6,22 @@ include ../../makefile.static
 CXXSRC = SISConeBasePlugin.cc SISConePlugin.cc SISConeSphericalPlugin.cc
 OBJS = $(patsubst %.cc,%.o,$(CXXSRC)) 
 
-MAKESUBDIR = siscone_config siscone_src
-SUBDIR = siscone/siscone
-SUBDIR_CONFIG = siscone
-
 # fastjet includes
 FASTJET_INCLUDE = -I../../include
 INCLUDE += $(FASTJET_INCLUDE)
 
 # scones includes
-INCLUDE += -I$(SUBDIR_CONFIG)
+SISCONE_INCLUDE = -I$(PREFIX)/include -I$(PREFIX)/include/siscone
+INCLUDE += $(SISCONE_INCLUDE)
 
 all: libSISConePlugin.a
 
-libSISConePlugin.a: $(MAKESUBDIR) $(OBJS)
-	ar cru libSISConePlugin.a $(OBJS) $(SUBDIR)/*.o
+libSISConePlugin.a: $(OBJS)
+	ar cru libSISConePlugin.a $(OBJS) -L$(PREFIX)/lib -lsiscone -lsiscone_spherical
 	ranlib libSISConePlugin.a
 
-siscone_config:
-	@cd $(SUBDIR_CONFIG) && $(MAKE) -f makefile.static config
-
-siscone_src:
-	@cd $(SUBDIR) && $(MAKE) -f makefile.static $(MAKECMDGOALS)
-
 clean:
 	rm -f $(OBJS)
-	@cd $(SUBDIR) && $(MAKE) -f makefile.static $(MAKECMDGOALS)
 
 distclean: clean
 	rm -f libSISConePlugin.a *~
@@ -39,7 +29,7 @@ distclean: clean
 # NB: only use "relative" includes here (others are too subject
 #     to change according to the system one is on...)
 depend:
-	makedepend  -f makefile.static $(FASTJET_INCLUDE) -I. -Isiscone/src --   -- $(CXXSRC)
+	makedepend  -f makefile.static $(FASTJET_INCLUDE) -I. $(SISCONE_INCLUDE) --   -- $(CXXSRC)
 
 # DO NOT DELETE
 
-- 
2.47.1

